### PR TITLE
Workflow: Migrate QEMU playbook check to github workflow

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

Migrated QEMU playbook check functionality. Relevant docs here https://github.com/uraimo/run-on-arch-action

The gh action does not support riscv, so that will still need to remain in QPC.

This yaml file is alot easier to maintain than the entire QPC jenkins job, and its alot easier to add new arch distro combos

### CODE REVIEW

File: .github/workflows/build_qemu.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            distro: ubuntu18.04
+          - arch: armv7
+            distro: jessie
+
+    steps:
+      - uses: actions/checkout@v2.1.0
+
+      - name: Run on architecture
+        uses: uraimo/run-on-arch-action@v2.3.0
+
```
Comment (gdams) [inline .github/workflows/build_qemu.yml:39]: remove the space here

File: .github/workflows/build_qemu.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          distro: ${{ matrix.distro }}
+      
+      - name: Install Ansible
+        run: |
+          apt -y update
+          apt -y install ansible
+
+      - name: Run the UNIX playbook
+        run: |
+        echo "localhost ansible_user=runner ansible_connection=local" > ansible/hosts
```
Comment (gdams) [inline .github/workflows/build_qemu.yml:51]: spacing is out (needs another tab)

**Comment by sxa** [comment]
Not sure this is the right approach to force a github action to do this on every PR - will this not end up in a situation where all changes will result in a significant (potentially 1 day+) turnaround for PR approvals ... And repeated for all subsequent pushes/review fixes? I'd personaly prefer to stick with the existing QPC implementation where it can be run by individuals as required, and also on a scheduled regular basis to check the state of the playbooks.

**Comment by gdams** [comment]
> Not sure this is the right approach to force a github action to do this on every PR - will this not end up in a situation where all changes will result in a significant (potentially 1 day+) turnaround for PR approvals 

@sxa why not trigger it by adding a label? you could do something like this:

```yaml
on:
  pull_request:
    types: [ labeled ]

jobs:
  build:
    if: ${{ github.event.label.name == 'run-playbook-checker' }}
```

**Comment by Haroon-Khel** [comment]
>It's an option but I'm not convinced there's a compelling advantage to running it that way unless we're fixated on running these things in GitHub

Its alot easier to add platforms; instead of creating a new disk, updating the QEMU bash script and the jenkins job, I can add 2 lines to the yaml file

**Comment by Haroon-Khel** [comment]
Thats annoying. When one fails the rest get cancelled mid execution

**Comment by Haroon-Khel** [comment]
Strange that it failed here 
```
  TASK [Debug : Display Debug Information] ***************************************
  fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible_default_ipv4' is undefined\n\nThe error appears to have been in '/home/runner/work/infrastructure/infrastructure/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Debug/tasks/main.yml': line 5, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n#####################\n- name: Display Debug Information\n  ^ here\n"}

```
Because all of the variables in debug default to undefined if they are undefined

**Comment by Haroon-Khel** [comment]
I'm of the opinion that the Swap and Crontab roles should not run on non adoptium machines. They make changes to the machine that only we at adoptium need. Anyone else feel this way?

**Comment by gdams** [comment]
> The gh action does not support riscv, so that will still need to remain in QPC.

@Haroon-Khel I've just pushed risc-v support to the action so you should be able to add that

**Comment by Haroon-Khel** [comment]
QEMU workflows got triggered when I added the label to the pr. Works as expected 👍🏻

**Comment by Haroon-Khel** [comment]
Looks like you have to remove and re add the label for subsequent pushes to the same pr

**Comment by Haroon-Khel** [comment]
@sxa for review. This pr is at a stage where it can be merged

**Review by sxa — APPROVED** [review]
As discussed we still need the full  functionality of QPC but I'm ok to let this to go in for now.

**Comment by sxa** [comment]
@Haroon-Khel Doesn't look like RISC-V has been added after the earlier comments.

**Review by steelhead31 — APPROVED** [review]
Looks good.

**Pull request merged**